### ISSUE TITLE
Changing nms threshold decreases accuracy and results in false negatives.

### ISSUE THREAD

**Issue by QuickLearner171998** [tensorflow/models#7105]
What is the top-level directory of the model you are using:NA
Have I written custom code (as opposed to using a stock example script provided in TensorFlow):NA
OS Platform and Distribution (e.g., Linux Ubuntu 16.04):NA
TensorFlow installed from (source or binary):NA
TensorFlow version (use command below):NA
Bazel version (if compiling from source):NA
CUDA/cuDNN version:NA
GPU model and memory:NA
Exact command to reproduce:NA

### Describe the problem
My model is giving multiple boxes for same object detected. So I am trying to change the nms_thresholds in the config file. Although now the problem is solved but this results in lots of objects not being detected and also decrease in accuracy. 
I have increased **first_stage_nms_score_threshold: 0.0 to 0.6 and first_stage_nms_iou_threshold to 0.5**. In second_stage_post_processing, **batch_non_max_suppression score_threshold: 0.8 ,iou_threshold: 0.5**
So **I need to know why are the detections being affected**. And secondly **please suggest a way to remove redundant boxes without affecting the accuracy and all objects being detected as earlier.**
### faster_rcnn_inception_v2_pets _for_testing_inference.config  :
Changes made:

model {
  faster_rcnn {
    num_classes: 2
    image_resizer {
      keep_aspect_ratio_resizer {
        min_dimension: 1024
        max_dimension: 1024
      }
    }
    feature_extractor {
      type: 'faster_rcnn_inception_v2'
      first_stage_features_stride: 16
    }
    first_stage_anchor_generator {
      grid_anchor_generator {
        scales: [0.25, 0.5, 1.0, 2.0]
        aspect_ratios: [0.5, 1.0, 2.0]
        height_stride: 16
        width_stride: 16
      }
    }
    first_stage_box_predictor_conv_hyperparams {
      op: CONV
      regularizer {
        l2_regularizer {
          weight: 0.0
        }
      }
      initializer {
        truncated_normal_initializer {
          stddev: 0.01
        }
      }
    }
    **first_stage_nms_score_threshold: 0.6
    first_stage_nms_iou_threshold: 0.5
    first_stage_max_proposals: 100**
    first_stage_localization_loss_weight: 2.0
    first_stage_objectness_loss_weight: 1.0
    initial_crop_size: 14
    maxpool_kernel_size: 2
    maxpool_stride: 2
    second_stage_box_predictor {
      mask_rcnn_box_predictor {
        use_dropout: false
        dropout_keep_probability: 1.0
        fc_hyperparams {
          op: FC
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            variance_scaling_initializer {
              factor: 1.0
              uniform: true
              mode: FAN_AVG
            }
          }
        }
      }
    }
    second_stage_post_processing {
      batch_non_max_suppression {
        _**score_threshold: 0.8
        iou_threshold: 0.5**_
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SOFTMAX
    }
    second_stage_localization_loss_weight: 2.0
    second_stage_classification_loss_weight: 1.0
  }
}

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
